File: mycuration/src/androidTest/java/com/phicdy/mycuration/util/test/FileUtilTest.java
package com.phicdy.mycuration.util.test;

import android.support.test.runner.AndroidJUnit4;

import com.phicdy.mycuration.util.FileUtil;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import static android.support.test.InstrumentationRegistry.getTargetContext;
import static junit.framework.Assert.assertEquals;

@RunWith(AndroidJUnit4.class)
public class FileUtilTest {

    @Before
    public void setup() {
        FileUtil.INSTANCE.setUpIconSaveFolder(getTargetContext());
    }

	@Test
	public void testGetIconSavePath() {
		assertEquals(FileUtil.INSTANCE.getAppPath(getTargetContext()) + "icons/",
				FileUtil.INSTANCE.iconSaveFolder());
	}

	@Test
	public void testGenerateIconFileName() {
		String iconSaveFolderStr = FileUtil.INSTANCE.iconSaveFolder();
		assertEquals(
				FileUtil.INSTANCE.getAppPath(getTargetContext()) + "icons/gigazine.net.png",
				FileUtil.INSTANCE.generateIconFilePath(iconSaveFolderStr, "http://gigazine.net/"));
	}
}

File: mycuration/src/androidTest/java/com/phicdy/mycuration/util/test/FileUtilTest.kt
package com.phicdy.mycuration.util.test

import androidx.test.core.app.ApplicationProvider
import androidx.test.ext.junit.runners.AndroidJUnit4
import com.phicdy.mycuration.util.FileUtil
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class FileUtilTest {

    @Before
    fun setup() {
        FileUtil.setUpAppPath(ApplicationProvider.getApplicationContext())
    }

    @Test
    fun testGetIconSavePath() {
        assertEquals(FileUtil.getAppPath(ApplicationProvider.getApplicationContext()) + "icons/", FileUtil.iconSaveFolder())
    }
}
